### PR TITLE
Common/CommonFuncs: Remove now-unneccessary ArraySize function

### PR DESCRIPTION
**Pull request by lioncash**
Since C++17, non-member [std::size()](https://en.cppreference.com/w/cpp/iterator/size) is present in the standard library which also operates on regular C arrays. Given that, we can just replace usages of ArraySize with that where applicable.

In many cases, we can just change the actual C array ArraySize() was called on into a `std::array` and use its `size()` member function instead. In some other cases, we can collapse the loops they were used in into a ranged-for loop, eliminating the need for en explicit bounds query.

### CODE REVIEW

File: Source/Core/Core/Boot/Boot.cpp [hunk 1]
```diff
@@ -255,9 +255,11 @@ bool CBoot::FindMapFile(std::string* existing_map_file, std::string* writable_ma
...
 
   bool found = false;
-  static const std::string maps_directories[] = {File::GetUserPath(D_MAPS_IDX),
-                                                 File::GetSysDirectory() + MAPS_DIR DIR_SEP};
-  for (size_t i = 0; !found && i < ArraySize(maps_directories); ++i)
+  static const std::array<std::string, 2> maps_directories{
+      File::GetUserPath(D_MAPS_IDX),
+      File::GetSysDirectory() + MAPS_DIR DIR_SEP,
+  };
+  for (size_t i = 0; !found && i < maps_directories.size(); ++i)
```
Comment (BhaaLseN) [inline Source/Core/Core/Boot/Boot.cpp]: Couldn't this be a ranged for-loop with an immediate `return true` inside the loop?

Comment (lioncash) [reply Source/Core/Core/Boot/Boot.cpp]: Amended the commit with that suggestion

File: Source/Core/Core/HLE/HLE.cpp [hunk 2]
```diff
@@ -41,7 +42,7 @@ struct SPatch
 };
 
 // clang-format off
-static const SPatch OSPatches[] = {
+constexpr std::array<SPatch, 21> OSPatches{{
```
Comment (Tilka) [inline Source/Core/Core/HLE/HLE.cpp]: Why not `static` anymore?

Comment (lioncash) [reply Source/Core/Core/HLE/HLE.cpp]: const (and by extension constexpr) objects already have internal linkage. static is unnecessary here.

**Pull request merged**